src/modules/rangeproof/rangeproof_impl.h

```diff
@@ -421,21 +585,11 @@ SECP256K1_INLINE static int secp256k1_rangeproof_rewind_inner(secp256k1_scalar *
         }
         return 0;
     }
-    skip1 = rsizes[rings - 1] - 1 - j;
-    skip2 = ((value >> ((rings - 1) << 1)) & 3);
-    if (skip1 == skip2) {
-        /*Value is in wrong position.*/
-        if (mlen) {
-            *mlen = 0;
-        }
-        return 0;
-    }
-    skip1 += (rings - 1) << 2;
-    skip2 += (rings - 1) << 2;
     /* Like in the rsize[] == 1 case, Having figured out which s is the one which was not forged, we can recover the blinding factor. */
-    secp256k1_rangeproof_recover_x(&stmp, &s_orig[skip2], &ev[skip2], &s[skip2]);
-    secp256k1_scalar_negate(&sec[rings - 1], &sec[rings - 1]);
-    secp256k1_scalar_add(blind, &stmp, &sec[rings - 1]);
+    final_x_pos = 4 * (header->n_rings - 1) + ((*v >> (2 * (header->n_rings - 1))) & 3);
```

Comment:
AFAICs `*v` isn't the same as `value`. If `!if ((tmp[0] & 128) && (memcmp(&tmp[16], &tmp[24], 8) == 0) && (memcmp(&tmp[8], &tmp[16], 8) == 0))` then `*v = 0` and `value` is `UINT_MAX`. And in the other places where we access `v` we check that it's not NULL.

Reply:
We should remove any checks that `v` is non-NULL. We have always been inconsistent about this will always have segfaulted if `v` were every NULL (but fortunately it never is).

In later commits (maybe not in this PR) I move away from the name `v` and toward `proven_value` vs `value`. You are correct that `*v` is not the same as `value`. I'm not sure what point you are making about it being 0 sometimes.

Reply:
I was wrong, `*v` is the same as `value` at this point.